product_docs/docs/pem/9/considerations/pem_pgbouncer/preparing_the_pem_database_server.mdx

```diff
@@ -8,54 +8,83 @@ redirects:
+
+## Prerequisites 
+
+-   You are connected to the `pem` database of the PEM database server. 
+
+-   You are connected as `enterprisedb` or `postgres` user. The user depends on your Postgres distribution.
+
+    | Postgres distribution        | User         |
+    |------------------------------|--------------|
+    | EDB Postgres Advanced Server | enterprisedb |
+    | EDB Postgres Extended Server | postgres     |
+    | PostgreSQL                   | postgres     |
+
+## Creating users and roles for PgBouncer-PEM connections
+
+1.  Create a dedicated user named pgbouncer with `pem_agent_pool` membership:
 
     ```sql
-    CREATE USER pgbouncer PASSWORD 'ANY_PASSWORD' LOGIN;
+    CREATE ROLE pgbouncer PASSWORD 'ANY_PASSWORD' LOGIN;
     __OUTPUT__
     CREATE ROLE
     ```
 
-2.  Create a user named pem_admin1 (not a superuser) with `pem_admin` and `pem_agent_pool role` membership on the PEM database server:
+    ```sql
+    GRANT pem_agent_pool TO pgbouncer;
+    __OUTPUT__
+    GRANT ROLE
+    ```
+
+1.  Create a user named pem_admin1 (not a superuser) with `pem_admin` and `pem_agent_pool role` membership:
 
     ```sql
-    CREATE USER pem_admin1 PASSWORD 'ANY_PASSWORD' LOGIN CREATEROLE;
+    CREATE ROLE pem_admin1 PASSWORD 'ANY_PASSWORD' LOGIN CREATEROLE;
     __OUTPUT__
     CREATE ROLE
     ```
+
     ```sql
-    GRANT pem_admin, pem_agent_pool TO pem_admin1;
+    GRANT pem_agent_pool TO pem_admin1;
     __OUTPUT__
     GRANT ROLE
     ```
 
-3.  Grant CONNECT privileges to the pgbouncer user on the `pem` database:
+    ```sql
+    GRANT pem_agent TO pem_admin1 WITH ADMIN OPTION;
+    __OUTPUT__
+    GRANT ROLE
+    ```
+
+1.  Grant CONNECT privileges to the pgbouncer user:
 
     ```sql
     GRANT CONNECT ON DATABASE pem TO pgbouncer;
     __OUTPUT__
     GRANT
     ```
 
-4.  Grant USAGE privileges to the pgbouncer user for the `pem` schema on the `pem` database:
+1.  Grant USAGE privileges to the pgbouncer user for the `pem` schema:
 
     ```sql
     GRANT USAGE ON SCHEMA pem TO pgbouncer;
     __OUTPUT__
     GRANT
     ```
 
-5.  Grant EXECUTE privileges to the pgbouncer user on the `pem.get_agent_pool_auth(text)` function in the `pem` database. For example:
+1.  Grant EXECUTE privileges to the pgbouncer user on the `pem.get_agent_pool_auth(text)` function. For example:
 
     ```sql
     GRANT EXECUTE ON FUNCTION pem.get_agent_pool_auth(text) TO pgbouncer;
     __OUTPUT__
     GRANT
     ```
 
-6.  Use the `pem.create_proxy_agent_user(varchar)` function to create a user named pem_agent_user1 on the PEM database server:
+1.  Use the `pem.create_proxy_agent_user(varchar)` function to create a user named pem_agent_user1:
```

Comment:
This is the third user/login role we're creating. What is this one for? How does it relate to the others?